packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -8,32 +8,51 @@ import 'dart:ui' as ui show Codec, FrameInfo, Image;
 import 'package:flutter/foundation.dart';
 import 'package:flutter/scheduler.dart';
 
-const String _flutterWidgetsLibrary = 'package:flutter/widgets.dart';
+const String _flutterPaintingLibrary = 'package:flutter/painting.dart';
 
 /// A [dart:ui.Image] object with its corresponding scale.
 ///
 /// ImageInfo objects are used by [ImageStream] objects to represent the
 /// actual data of the image once it has been obtained.
 ///
-/// The receiver of an [ImageInfo] object must call [dispose]. To safely share
-/// the object with other clients, use the [clone] method before calling
-/// dispose.
+/// The receiver of an [ImageInfo] object must call [dispose].
 @immutable
 class ImageInfo {
   /// Creates an [ImageInfo] object for the given [image] and [scale].
   ///
+  /// The created [ImageInfo] will dispose the [image] when [dispose] is called.
+  /// If the [image] needs to be shared with other clients, pass `image.clone()`
+  /// The cloning will not increase memory footprint, because it clones
+  /// just handler to the image.
+  ///
+  /// After passing [image] to the constructor,
+  /// remove any references to it so that the image can be garbage collected,
+  /// after ImageInfo disposal.
+  ///
   /// The [debugLabel] may be used to identify the source of this image.
-  const ImageInfo({ required this.image, this.scale = 1.0, this.debugLabel });
+  ///
+  /// See also:
+  ///
+  ///  * [Image.clone], which describes how and why to clone images.
+  ImageInfo({
+    required this.image,
+    this.scale = 1.0,
+    this.debugLabel,
+  })
+  {
+    if (kFlutterMemoryAllocationsEnabled) {
+      MemoryAllocations.instance.dispatchObjectCreated(
+        library: _flutterPaintingLibrary,
+        className: '$ImageInfo',
+        object: this,
+      );
+    }
+  }
 
   /// Creates an [ImageInfo] with a cloned [image].
   ///
-  /// Once all outstanding references to the [image] are disposed, it is no
-  /// longer safe to access properties of it or attempt to draw it. Clones serve
-  /// to create new references to the underlying image data that can safely be
-  /// disposed without knowledge of whether some other reference holder will
-  /// still need access to the underlying image. Once a client disposes of its
-  /// own image reference, it can no longer access the image, but other clients
-  /// will be able to access their own references.
+  /// Cloning of the image will not increase memory foot print, because it clones
```

Comment:
This isn't quite true, and I'm not sure why we're deleting the existing documentation in favor of this...